.github/workflows/flutter-desktop-tests.yml

```diff
@@ -42,26 +38,33 @@ jobs:
         with:
           submodules: false
 
+      - name: Check runner.arch
+        if: ${{ inputs.arch != runner.arch }}
+        run: false # fail if arch is not as expected
+        shell: bash
+
+      - name: Set RUNNER_OS
+        # there is no such thing as ${{ tolower(runner.os) }}, hence this abomination ¯\_(ツ)_/¯ 
+        run: echo ${{ runner.os }} | awk '{print "RUNNER_OS=" tolower($0)}' >> $GITHUB_ENV
```

Comment:
[nit] Rather than set is as an env variable, a more idiomatic approach would be to set it as the output of the step and use it later as `steps.set-runner-os.outputs.os` - it's more verbose, but it's nicer than stashing everything in env.